scripts/rhel_validation/take_kernel_version.ps1

```diff
@@ -0,0 +1,166 @@
+################################################################################
+#
+# Linux on Hyper-V and Azure Test Code, ver. 1.0.0
+# Copyright (c) Microsoft Corporation
+#
+# All rights reserved.
+# Licensed under the Apache License, Version 2.0 (the ""License"");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS
+# OF ANY KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION
+# ANY IMPLIED WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR
+# PURPOSE, MERCHANTABLITY OR NON-INFRINGEMENT.
+#
+# See the Apache Version 2.0 License for specific language governing
+# permissions and limitations under the License.
+#
+################################################################################
+
+################################################################################
+#
+#	Description
+#
+#	This script will find out if new kernel exists for each version of Redhat
+# from RHEL6.0 to RHEL7.4.
+#	This script imports the login cookies for downloading the html to get 
+# the entire list of kernels. For each version of Redhat it creates a list with 
+# kernels associated and stores these informations in a hash table.
+#	Each version of Redhat has a latest kernel stored in a file which is compared 
+# with the last added kernel in hash table.
+#
+################################################################################
+################################################################################
+#
+# Main script body
+#
+################################################################################
+
+param (
+    [String] $WorkDir,
+    [String] $LatestVersionFile,
+    [String] $KernelVersionsPath,
+    [String] $UtilsDir
+)
+
+if (!(Test-Path $WorkDir)) {
+    New-Item -Type Directory -Path $WorkDir
+}
+if (Test-Path "$WorkDir\package.html") {
+    Remove-Item  "$WorkDir\package.html"
+}
+New-Item -Path $KernelVersionsPath -Force
+$KernelVersionsPath = Resolve-Path $KernelVersionsPath
+
+if (Test-Path $LatestVersionFile) {
+    $LatestVersions = Get-Content $LatestVersionFile
+    $LatestVersions = $LatestVersions.Split(";")
+    foreach ($entry in $LatestVersions) {
+        if ($entry) {
+            $distro = $entry.split("=")[0]
+            $kernel = $entry.split("=")[1]
+            $LatestVersionsHash += @{$distro = $kernel}
+        }
+    }
+} else {
+    Write-Host "No versions file found. Creating a new one"
+    New-Item -Path $LatestVersionFile -Force
+    $LatestVersionsHash = @{}
+}
+Copy-Item "$UtilsDir\cookies_redhat.json" $WorkDir
+
+# generate hash table with list of kernels for each version of rhel
+$hash = @{"rhel_7.4" = @{"baseVer" = "3.10.0-693"; "newVer" = @()}; 
```

Comment:
this hash is a static variable and should be put at the beginning and called RHEL_VERSIONS_TO_KERNEL_MAP